src/main/java/com/iota/iri/service/snapshot/impl/SnapshotStateImpl.java

```diff
@@ -87,7 +49,7 @@ public SnapshotStateImpl(SnapshotState snapshotState) {
      *
      * @param balances map with the addresses associated to their balance
      */
-    private SnapshotStateImpl(Map<Hash, Long> balances) {
+    protected SnapshotStateImpl(Map<Hash, Long> balances) {
```

Comment:
Because before we created the SnapshotStates from within the class (fromFile method) and now the fromFile method is part of another class (the SnapshotProvider) since the model should be independant from the persistance layer.